app/controllers/races/bingos_controller.rb

```diff
@@ -6,6 +6,6 @@ def show
   private
 
   def set_race
-    @race = Bingo.find(params[:race])
+    @race = Bingo.where('LEFT(id::text, ?) = ?', race_params[:race].length, race_params[:race]).order(created_at: :asc).first
```

Comment:
Might be a good idea to move this to some shared method so that each controller can use the same lookup.